### ISSUE TITLE
[Bug]: ut test TestDeadLockWith2Txn fail

### ISSUE THREAD

**Issue by YANGGMM** [matrixorigin/matrixone#11827]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

<img width="539" alt="image" src="https://github.com/matrixorigin/matrixone/assets/26563383/6bbe1f48-8265-436a-99a0-4302fcf622fe">

fail link：https://github.com/matrixorigin/matrixone/actions/runs/6230030980/job/16909466795?pr=11770

**Comment by zhangxu19830126** [comment]
未处理

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by xzxiong** [comment]
meet again
- commit id: f43fcb6ec
- date: 2023-10-08
- link, https://github.com/matrixorigin/matrixone/actions/runs/6446562824
```
2023-10-08T08:50:27.4855020Z # FAILED CASES:
2023-10-08T08:50:27.4855360Z --- FAIL: TestDeadLockWith2Txn (0.26s)
2023-10-08T08:50:27.4855510Z
```
- stack
```
2023-10-08T08:44:19.5306130Z ==================
2023-10-08T08:44:19.5306680Z WARNING: DATA RACE
2023-10-08T08:44:19.5306950Z Write at 0x00c000984f50 by goroutine 5230:
2023-10-08T08:44:19.5307330Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).reset()
2023-10-08T08:44:19.5307880Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/waiter.go:249 +0x93
2023-10-08T08:44:19.5308310Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).close()
2023-10-08T08:44:19.5308830Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/waiter.go:117 +0x4f
2023-10-08T08:44:19.5309290Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*sliceBasedWaiterQueue).notify()
2023-10-08T08:44:19.5309860Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/waiter_queue.go:96 +0x2a8
2023-10-08T08:44:19.5310300Z   github.com/matrixorigin/matrixone/pkg/lockservice.Lock.closeTxn()
2023-10-08T08:44:19.5311180Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/lock.go:135 +0x3e6
2023-10-08T08:44:19.5311640Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*localLockTable).unlock.func1()
2023-10-08T08:44:19.5312230Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/lock_table_local.go:199 +0x40b
2023-10-08T08:44:19.5312660Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*fixedSlice).iter()
2023-10-08T08:44:19.5313510Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/slice.go:159 +0xd6
2023-10-08T08:44:19.5313950Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*localLockTable).unlock()
2023-10-08T08:44:19.5314540Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/lock_table_local.go:188 +0x2ce
2023-10-08T08:44:19.5314980Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*activeTxn).close()
2023-10-08T08:44:19.5315500Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/txn.go:139 +0x30d
2023-10-08T08:44:19.5315920Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*service).Unlock()
2023-10-08T08:44:19.5316460Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service.go:146 +0x31e
2023-10-08T08:44:19.5316940Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1.1.1()
2023-10-08T08:44:19.5317550Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1053 +0x1b8
2023-10-08T08:44:19.5465780Z
2023-10-08T08:44:19.5466130Z Previous read at 0x00c000984f50 by goroutine 5231:
2023-10-08T08:44:19.5466570Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).String()
2023-10-08T08:44:19.5467140Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/waiter.go:88 +0x4c
2023-10-08T08:44:19.5468320Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).notify()
2023-10-08T08:44:19.5468870Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/waiter.go:218 +0x9d
2023-10-08T08:44:19.5469320Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*activeTxn).cancelBlocks()
2023-10-08T08:44:19.5469870Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/txn.go:183 +0x17b
2023-10-08T08:44:19.5470310Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*activeTxn).close()
2023-10-08T08:44:19.5470840Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/txn.go:118 +0x7d
2023-10-08T08:44:19.5471250Z   github.com/matrixorigin/matrixone/pkg/lockservice.(*service).Unlock()
2023-10-08T08:44:19.5471800Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service.go:146 +0x31e
2023-10-08T08:44:19.5472290Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1.1.2()
2023-10-08T08:44:19.5472900Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1059 +0x1b8
2023-10-08T08:44:19.5473120Z
2023-10-08T08:44:19.5473250Z Goroutine 5230 (running) created at:
2023-10-08T08:44:19.5473680Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1.1()
2023-10-08T08:44:19.5474270Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1049 +0x775
2023-10-08T08:44:19.5474710Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1.1()
2023-10-08T08:44:19.5475660Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:70 +0x61a
2023-10-08T08:44:19.5475960Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel.func1()
2023-10-08T08:44:19.5476330Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1322 +0x1da
2023-10-08T08:44:19.5476580Z   github.com/matrixorigin/matrixone/pkg/lockservice.RunLockServicesForTest()
2023-10-08T08:44:19.5476940Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/test_helper.go:76 +0xaaa
2023-10-08T08:44:19.5477220Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel()
2023-10-08T08:44:19.5477580Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1313 +0x10e
2023-10-08T08:44:19.5478180Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTests()
2023-10-08T08:44:19.5478550Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1281 +0x112
2023-10-08T08:44:19.5478790Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1()
2023-10-08T08:44:19.5479140Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:47 +0x53
2023-10-08T08:44:19.5479740Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1()
2023-10-08T08:44:19.5480100Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1032 +0xb3
2023-10-08T08:44:19.5480200Z   testing.tRunner()
2023-10-08T08:44:19.5480560Z       /Users/runner/hostedtoolcache/go/1.20.8/x64/src/testing/testing.go:1576 +0x216
2023-10-08T08:44:19.5480680Z   testing.(*T).Run.func1()
2023-10-08T08:44:19.5481020Z       /Users/runner/hostedtoolcache/go/1.20.8/x64/src/testing/testing.go:1629 +0x47
2023-10-08T08:44:19.5481040Z
2023-10-08T08:44:19.5481160Z Goroutine 5231 (running) created at:
2023-10-08T08:44:19.5481440Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1.1()
2023-10-08T08:44:19.5481800Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1055 +0x99d
2023-10-08T08:44:19.5482040Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1.1()
2023-10-08T08:44:19.5482390Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:70 +0x61a
2023-10-08T08:44:19.5482700Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel.func1()
2023-10-08T08:44:19.5483050Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1322 +0x1da
2023-10-08T08:44:19.5483630Z   github.com/matrixorigin/matrixone/pkg/lockservice.RunLockServicesForTest()
2023-10-08T08:44:19.5483990Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/test_helper.go:76 +0xaaa
2023-10-08T08:44:19.5484270Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel()
2023-10-08T08:44:19.5484630Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1313 +0x10e
2023-10-08T08:44:19.5484880Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTests()
2023-10-08T08:44:19.5485230Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1281 +0x112
2023-10-08T08:44:19.5485460Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1()
2023-10-08T08:44:19.5485810Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:47 +0x53
2023-10-08T08:44:19.5486080Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestDeadLockWith2Txn.func1()
2023-10-08T08:44:19.5486440Z       /Users/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1032 +0xb3
2023-10-08T08:44:19.5486560Z   testing.tRunner()
2023-10-08T08:44:19.5486900Z       /Users/runner/hostedtoolcache/go/1.20.8/x64/src/testing/testing.go:1576 +0x216
2023-10-08T08:44:19.5487020Z   testing.(*T).Run.func1()
2023-10-08T08:44:19.5487370Z       /Users/runner/hostedtoolcache/go/1.20.8/x64/src/testing/testing.go:1629 +0x47
2023-10-08T08:44:19.5487780Z ==================
```

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by zhangxu19830126** [comment]
fix by #12090
